p8105\_hw1\_tk2886
================
Tanvir Khan
2021-09-24

## QUESTION 1

``` r
set.seed(12345)

mydataframe <- tibble(
                  random_sample = rnorm(10),
                  logical_vec = c(random_sample > 0),
                  char_vec = c("My", "name", "is", "Tanvir", "Khan", "and", "I", "like","the", "Lakers"),
                  fac_vec = factor(c("single", "single", "single", "single", "married", "married", "married", "divorced", "divorced", "divorced")),
)

knitr::kable(mydataframe)
```

| random\_sample | logical\_vec | char\_vec | fac\_vec |
|---------------:|:-------------|:----------|:---------|
|      0.5855288 | TRUE         | My        | single   |
|      0.7094660 | TRUE         | name      | single   |
|     -0.1093033 | FALSE        | is        | single   |
|     -0.4534972 | FALSE        | Tanvir    | single   |
|      0.6058875 | TRUE         | Khan      | married  |
|     -1.8179560 | FALSE        | and       | married  |
|      0.6300986 | TRUE         | I         | married  |
|     -0.2761841 | FALSE        | like      | divorced |
|     -0.2841597 | FALSE        | the       | divorced |
|     -0.9193220 | FALSE        | Lakers    | divorced |

``` r
rnorm_mean = mean(pull(mydataframe, random_sample))
logicalvec_mean = mean(pull(mydataframe, logical_vec))
charvec_mean = mean(pull(mydataframe, char_vec))
## Warning in mean.default(pull(mydataframe, char_vec)): argument is not numeric or
## logical: returning NA
facvec_mean = mean(pull(mydataframe, fac_vec))
## Warning in mean.default(pull(mydataframe, fac_vec)): argument is not numeric or
## logical: returning NA
```

When I take the mean of the random sample of size 10 from a standard
Normal distribution, the mean function works on this and it gives me a
value. This works because it contains numeric value.

When I take the mean of the logical vector, the mean functions work on
this logical vector even though that does not make sense. If someone
looks at the dataframe they will see that the logical vector in my
dataframe displays True and False. The True is being converted to 1 and
False is being converted to 0. I have four True out of 10 and that is
0.4 which is the same value as the mean function is outputting.

When I take the mean of the character vector and factor vector, the
console output is displaying a warning message that states the argument
is not numeric or logical: “Returning NA”. This makes sense for the
character vector since I can not take the mean of a character vector
which does not have numeric values. The factor vector is not numeric.

``` r
numeric_char = as.numeric(factor(pull(mydataframe,char_vec)))
numeric_logv = as.numeric(pull(mydataframe, logical_vec))
numeric_fac = as.numeric(pull(mydataframe, fac_vec))
```

``` r
mean_numchar = mean(numeric_char)
mean_numlogic = mean(numeric_logv)
mean_numfactor = mean(numeric_fac)
```

## Question 2

``` r
data("penguins", package = "palmerpenguins")
```

1.  The important variables are species, island, bill\_length\_mm,
    bill\_depth\_mm, flipper\_length\_mm, body\_mass\_g, sex, year.
2.  There is 344 rows. There is 8 columns.
3.  The mean of the flipper length is 201.

``` r
penguin_scatterplot <- ggplot(penguins, aes(x = bill_length_mm, y = flipper_length_mm, color = species)) +
  geom_point(size = 2, shape = 23)

penguin_scatterplot
```

    ## Warning: Removed 2 rows containing missing values (geom_point).

![](p8105_hw1_tk2886_files/figure-gfm/Creating%20scatterplot-1.png)<!-- -->

``` r
ggsave("scatter_plot.pdf", plot = penguin_scatterplot, height = 4, width = 6)
```

    ## Warning: Removed 2 rows containing missing values (geom_point).
